src/lib.rs

```diff
@@ -65,9 +59,8 @@ pub use crate::readonly::{
 pub use crate::sum_products::{dot, einsum_impl, inner};
 pub use ndarray::{array, Ix1, Ix2, Ix3, Ix4, Ix5, Ix6, IxDyn};
 
-/// Test readme
-#[doc(hidden)]
-pub mod doc_test {
+#[cfg(doc)]
```

Comment:
Might actually want to be

```suggestion
#[cfg(test)]
```

?

Reply:
I tried but it appears that `cargo test --doc` does not set `--cfg test`, so this would not work in the sense of detecting errors in the README that way.

Reply:
Ahhh, looks like it's `#[cfg(doctest)]` - https://github.com/PyO3/pyo3/blob/2cee7feaaf216e4d4cf87c855486ef01a17d0ffc/src/lib.rs#L371

Reply:
That one did the trick!